plugins/catalog-backend/src/database/Database.ts

```diff
@@ -367,18 +372,65 @@ export class Database {
     }
   }
 
-  async location(id: string): Promise<DbLocationsRow> {
-    const items = await this.database<DbLocationsRow>('locations')
-      .where({ id })
-      .select();
+  async location(id: string): Promise<DbLocationsRowWithStatus> {
+    const items = await this.database<DbLocationsRowWithStatus>('locations')
+      .where('locations.id', id)
+      .leftJoin(
+        'location_update_log',
+        'locations.id',
+        'location_update_log.location_id',
+      )
+      .orderBy('location_update_log.created_at', 'desc')
+      .select({
+        status: 'location_update_log.status',
+        timestamp: 'location_update_log.created_at',
+        message: 'location_update_log.message',
+        id: 'locations.id',
+        type: 'locations.type',
+        target: 'locations.target',
+      });
+
     if (!items.length) {
       throw new NotFoundError(`Found no location with ID ${id}`);
     }
     return items[0];
   }
 
-  async locations(): Promise<DbLocationsRow[]> {
-    return this.database<DbLocationsRow>('locations').select();
+  async locations(): Promise<DbLocationsRowWithStatus[]> {
+    const query = this.database
+      .select({
+        status: 'location_update_log.status',
+        timestamp: 'location_update_log.created_at',
+        message: 'location_update_log.message',
+        id: 'locations.id',
+        type: 'locations.type',
+        target: 'locations.target',
+      })
+      .from('locations')
+      .leftJoin(
+        'location_update_log',
+        'locations.id',
+        'location_update_log.location_id',
+      )
+      .orderBy('location_update_log.created_at', 'desc');
```

Comment:
If anybody has a better idea on this query - I'm open to suggestions ;)

Reply:
Are we looking to get all the location log entries or just the latest one? It might be useful to have a view into the location log table that only contains the latest update.

Reply:
Yes, the idea is for the list of locations show only last result per location. But the /history will contain last 10(?) updates. 
I like the idea with the predefined view for this purpose 👍

Reply:
agreed, a view might help here - or a subselect that has a limit(1) would work too

Reply:
- [x] create separate view for latest update